java-checks/src/test/files/checks/security/HostnameVerifierImplementationCheck.java

```diff
@@ -0,0 +1,72 @@
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLSession;
+
+class A {
+  private B b;
+
+  A(B b) {
+    this.b = b;
+  }
+
+  void method1() {
+    return b.method(new HostnameVerifier() {
+      @Override
+      public boolean verify(String a, SSLSession b) {
+        return true; // Noncompliant {{Do not unconditionally return true in this method.}}
+      }
+    });
+  }
+
+  void method2() {
+    return b.method((a, b) -> true); // Noncompliant
+  }
+
+  void method3() {
+    return b.method((a, b) -> (true)); // Noncompliant
```

Comment:
What about lambda : (a, b) -> (((true))) or return (((true))); 
I would add more than just one parenthesis.

Reply:
I am not sure we need to put those cases in different methods (a bit more compact testcase could not hurt).